### ISSUE TITLE
ntex::http::client::Client::post always return 400

### ISSUE THREAD

**Issue by xorxsan** [ntex-rs/ntex#123]
Hi,

   For some reason when sending a POST request to a server, the response is always 400 with the following text:
```html
<html>
    <head><title>400 Bad Request</title></head>
    <body>
    <center><h1>400 Bad Request</h1></center>
    </body>
</html>
```
Consider this code:
```toml
[package]
name = "client"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
ntex = { version = "=0.5.20", features = ["tokio", "rustls"] }
serde = "*"
serde_json = "*"
```
```rs

#[derive(serde::Serialize)]
struct MyRequest {
    a_value: String,
}

#[ntex::main]
async fn main() -> std::io::Result<()> {
    let client = ntex::http::client::ClientBuilder::default().finish();
    match client.post("https://httpbin.org/post")
        .send_json(&MyRequest {
            a_value: String::from("hola"),
        })
        .await {
        Ok(mut response) => {
            if response.status() != ntex::http::StatusCode::OK {
                println!("Error. Server responded with status code '{}' and error '{}'",
                         response.status(),
                         response.body()
                             .await
                             .unwrap()
                             .escape_ascii()
                             .to_string());
            }
            match response.body().await {
                Ok(bytes) => {
                    println!("{}", bytes.escape_ascii().to_string())
                },
                Err(error) => {
                    println!("Error: {}", error.to_string());
                }
            }
        }
        Err(error) => {
            println!("Error sending request: {}", error.to_string());
        }
    }
    Ok(())
}
```

With `ntex` version `0.5.20` this fails with a 400, however, with previous versions (like `0.5.19`) it works as expected.

https://httpbin.org/post is an echo server, but it happens with other servers.

I'm on Windows 10 x64 with
```sh
~ $ cargo --version
cargo 1.61.0 (a028ae42f 2022-04-29)
```

Thanks!

**Comment by leon3s** [comment]
Hey thank for your report!
I could repoduce this on ubuntu 22 and windows 11 with openssl and rustls
but it work fine if you replace https to http
So it's seems to be a SSL/TLS problem i'm investigating

**Comment by fafhrd91** [comment]
fixed in 0.5.21 release
